client/src/components/R&R/ProductBreakdown.jsx

```diff
@@ -1,8 +1,17 @@
+/* eslint-disable guard-for-in */
 import React from 'react';
+import FactorSlider from './FactorSlider';
 
-export default function ProductBreakdown() {
+export default function ProductBreakdown({ characteristics }) {
+  const sliders = [];
+
+  for (const characteristic in characteristics) {
+    const { value, id } = characteristics[characteristic];
+    sliders.push(<FactorSlider name={characteristic} value={value} key={id} />);
+  }
   return (
-    <div>
+    <div className="rr-product-breakdown">
+      {sliders}
     </div>
```

Comment:
i'm unfamiliar with this syntax? sliders is an array of components, so will it automatically render  all the components in this array?

Reply:
Exactly.  Just like when you do Array.map() to create an array of components.  Same idea.